projects/core/src/lib/map/map-layer.component.ts

```diff
@@ -311,12 +320,16 @@ export class OlRasterLayerComponent
 
         this.source.setTileLoadFunction((tile, src) => {
             const client = new XMLHttpRequest();
-
-            const sub = this.projectService.createQueryAbortStream().subscribe(() => {
-                (tile as any).getImage().src = 'assets/images/empty_tile.png';
-                client.abort();
-                sub.unsubscribe();
-                this.source.dispatchEvent('tileloaderror');
+            const tileZoomLevel: number = (tile as any).getTileCoord()[0];
+            const tileResolution = this.mapService.getView().getResolutionForZoom(tileZoomLevel);
+
+            const sub = this.projectService.createQueryAbortStream().subscribe((newResolution) => {
+                if (newResolution !== tileResolution) {
+                    (tile as any).setState(TileState.EMPTY);
+                    client.abort();
+                    sub.unsubscribe();
+                    this.source.dispatchEvent('tileloadend');
+                }
```

Comment:
In the documentation of OpenLayers, it states some slightly different behavior of custom loading, where it sets the error state at error and ok at… ok.

```js
import TileState from 'ol/TileState';

source.setTileLoadFunction(function(tile, src) {
  const xhr = new XMLHttpRequest();
  xhr.responseType = 'blob';
  xhr.addEventListener('loadend', function (evt) {
    const data = this.response;
    if (data !== undefined) {
      tile.getImage().src = URL.createObjectURL(data);
    } else {
      tile.setState(TileState.ERROR);
    }
  });
  xhr.addEventListener('error', function () {
    tile.setState(TileState.ERROR);
  });
  xhr.open('GET', src);
  xhr.send();
});
```

Maybe adapting this helps.

There is also the event `'abort'`. Maybe you can call abort manually if needed but react on all kind of abortions by using the default handler.

https://openlayers.org/en/latest/apidoc/module-ol_Tile.html#~LoadFunction

Reply:
We use the `tileloadend` event for the loading indicator.

Reply:
How does the loading indicator get this xhr object?

Moreover, we could just implement loaded, error and abort and handle the tile stuff there.

In the subscription, we just need to call abort.

Reply:
I meant that we count the number of `tileloadevent`s etc. to determine the status of the loading indicator. How will we determine that if we no longer dispatch the event?

Reply:
I did not say don't dispatch an event if this is necessary.
Just refactor the handling according to the docs and add `abort` if possible.

Reply:
I refactored it. I tried setting the TileState to EMPTY in a `addEventListener('abort')` method, but I got a tile status sequence violation error. As we do not set the state anywhere else, I assume this interferes with the internal state update. So I left it out. It works fine for me like this.

Reply:
I tried around, the thing with the sequence violation error is quite easy to grasp because each state is a number and you must not set a lower state.

I tried this:

```typescript
    // TileState.IDLE = 0
            // TileState.LOADING = 1
            // TileState.LOADED = 2
            // TileState.ERROR = 3
            // TileState.EMPTY = 4

            // tile.setState(TileState.LOADING);

            const debug_id = Math.random();

            client.addEventListener('loadend', (_event) => {
                console.log('loadend', debug_id);
                console.log('[1] tile.getState()', tile.getState());

                cancelSub.unsubscribe();
                const data = client.response;

                if (tile.getState() === TileState.EMPTY) {
                    console.log(data);
                }

                if (tile.getState() === TileState.EMPTY) {
                    return; // tile was aborted
                }

                if (!data) {
                    tile.setState(TileState.ERROR);
                } else {
                    (tile as any).getImage().src = URL.createObjectURL(data);
                    tile.setState(TileState.LOADED);
                }

                console.log('[2] tile.getState()', tile.getState());
            });
            client.addEventListener('error', () => {
                console.log('error', debug_id);
                console.log('tile.getState()', tile.getState());

                tile.setState(TileState.ERROR);
            });
            client.addEventListener('abort', () => {
                console.log('abort', debug_id);
                console.log('tile.getState()', tile.getState());

                tile.setState(TileState.EMPTY);
            });
            client.send();
        });
```

My insights:
1. After abort, there will no matter what be loadend called. Thus, you have to check whether to set ERROR then, because 3 < 4.
2. When you set LOADING at the beginning, our loading indicator will never stop (did not investigate this further)
3. When you set the tile to EMPTY, the loading indicator also won't stop loading (maybe please also check this)
4. For whatever reasons it seems to be fine to set nothing if everything goes fine :shrug: .
5. I guess you have to set TileState.ERROR on ERROR or ABORT. This is what you currently do, without knowing I guess, since the expectation was that we only set ERROR on error.

Reply:
makes sense. So I tested the loading indicator and it works when tiles are loaded normally, aborted, and errors are thrown. So it seems to me we can resolve this.